examples/expressions/13_timeseries_forecasting.py

```diff
@@ -0,0 +1,488 @@
+
+
+import polars as pl
+
+from skrub import TableReport, datasets
+
+data = datasets.fetch_bike_sharing().bike_sharing
+data = pl.from_pandas(data)
+TableReport(data)
+
+
+# %%
+# Prediction with datetime features
+# ---------------------------------
+#
+# In this section, we will use skrub Expressions to build a predictive pipeline
+# that pre-processes the data and performs hyperparameter optimization for various
+# of the steps involved in the preparation.
+# We will use a |RidgeCV| model as our learner. While it is not the most powerful
+# model, it will help us illustrate how feature engineering affects the prediction
+# performance.
+
+
+from sklearn.impute import SimpleImputer
+from sklearn.linear_model import RidgeCV
+
+###############################################################################
+# Since we are working with timeseries, we need to use |TimeSeriesSplit| to perform
+# crossvalidation.
+from sklearn.model_selection import TimeSeriesSplit
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+
+ts = TimeSeriesSplit(
+    n_splits=3,  # to keep the notebook fast enough on common laptops
+    gap=48,  # 2 days (48 hours) data gap between train and test
+    max_train_size=10000,  # keep train sets of comparable sizes
+    test_size=3000,  # for 2 or 3 digits of precision in scores
+)
+
+import skrub
+from skrub import TableVectorizer
+
+###############################################################################
+# We define a skrub |var| to start with.
+
+
+data_var = skrub.var("data", data)
+
+
+# We extract our input data (``X``) and the target column (``y``), and mark them
+# as X and y.
+
+
+X = data_var[
+    "date", "holiday", "weathersit", "temp", "hum", "windspeed"
+].skb.mark_as_X()
+y = data_var["cnt"].skb.mark_as_y()
+X
+
+
+# %%
+# We can use the |Cleaner| to clean the data and convert the date column to a
+# datetime column, as well as performing additional consistency and cleaning checks.
+
+
+from skrub import Cleaner
+
+X = X.skb.apply(Cleaner())
+
+# %%
+# Now we define a simple default pipeline for the |RidgeCV| predictor, which includes
+# a |StandardScaler| for numerical features and a |SimpleImputer| to handle
+# missing values. We will not modify this pipeline: we will instead focus on
+# pre-processing and feature engineering.
+
+
+default_ridge_pipeline = make_pipeline(StandardScaler(), SimpleImputer(), RidgeCV())
+
+
+# The "base" variant of the pipeline uses the default |TableVectorizer|, with
+# no hyperparameter search.
+
+
+vectorized = X.skb.apply(TableVectorizer())
+predictions_base = vectorized.skb.apply(default_ridge_pipeline, y=y)
+search_base = predictions_base.skb.get_grid_search(fitted=True, cv=ts)
+
+# %%
+# We can observe the results directly using ``.detailed_results``. Unsurprisingly,
+# the results are not very good. In this case, performing a GridSearch is not
+# necessary, but we will do it for consistency with the next steps.
```

Comment:
I'm confused here. We only search for the `alpha` of the `RidgeCV` but this will happen if we get the estimator since it is embedded in the learner itself, right? From the output, I see a single line, so it would be mean that we don't have really have a grid-search using a `GridSearchCV`.